### PR TITLE
kirkstone: updates for release v2024.12

### PR DESCRIPTION
**Pull request by lluiscampos**
Bringing from `scarthgap` #2218 and few other things that were pending.

### CODE REVIEW

**Comment by lluiscampos** [comment]
FYI @TheYoctoJester, I didn't bring #2178 for considering it of low value for `kirkstone`. Do you think we should port it?

File: meta-mender-core/recipes-mender/mender-client/mender_4.x.inc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+# The docker module depends on bash, and of course on docker. However, docker is
+# a very large requirement, which we will not mandate. Bash however, we require,
+# because otherwise the Yocto QA checks will complain.
+RDEPENDS:mender-update:append = "${@bb.utils.contains('PACKAGECONFIG', 'modules', ' bash', '', d)}"
```
Comment (lluiscampos) [inline meta-mender-core/recipes-mender/mender-client/mender_4.x.inc:5]: Note to self,

On `kirkstone`, this needs to be added for 3.x recipes too.

File: meta-mender-core/recipes-mender/mender-connect/mender-connect.inc [hunk 2]
```diff
@@ -13,19 +13,13 @@ SYSTEMD_SERVICE:${PN} = "mender-connect.service"
 
 B = "${WORKDIR}/build"
 
-inherit go
+inherit go-mod
```
Comment (lluiscampos) [inline meta-mender-core/recipes-mender/mender-connect/mender-connect.inc:16]: Note to self,

This change is the culprit of the CI failures. Changing the compile steps has added two dependencies on `mender-connect`:
* go-runtime >= 1.17.13
* libstd.so()(64bit)

Which increased the size of the final image making it not fit for one of the configurations.

To investigate:
* Why?
* Does it affect the same way `scarthgap` builds?
* Should we revert or just increase the image size?

Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-connect/mender-connect.inc:16]: See:
* https://github.com/mendersoftware/meta-mender/pull/2221

File: meta-mender-core/recipes-mender/mender-client/mender_4.x.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-core/recipes-mender/mender-client/mender_4.x.inc:5]: False alarm, 3.x recipes use `mender-client-go.inc` which was not modified (iow has this depenendy).

**Review by danielskinstad — APPROVED** [review]

**Comment by lluiscampos** [comment]
For reference, the build with Mender 3.7.8 and updating `kirkstone-next` was @ [Pipeline 1578583510](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1578583510)

**Pull request merged**